pytorch_lightning/trainer/data_loading.py

```diff
@@ -212,6 +241,19 @@ def __init__(self, num_features, dataset, *args, **kwargs):
 
         if not contains_dataset:
             dl_args.pop('dataset')
+        else:
+            # wrap the ``IterableDataset`` into a ``CaptureIterableDataset`` to record sampler states.
+            if fault_tolerant_enabled() and isinstance(dl_args["dataset"], IterableDataset):
+                # force the seed in the ``Dataset``
+                seed = int(os.getenv("PL_GLOBAL_SEED", 0)) + self.current_epoch + self.global_rank
+
+                if dl_args.get("generator") is None:
+                    dl_args["generator"] = torch.Generator().manual_seed(seed)
+
+                dl_args["dataset"] = CaptureIterableDataset(
+                    dataset=dl_args["dataset"],
+                    initial_seed=dl_args["generator"].initial_seed(),
+                )
```

Comment:
need to be careful here
loading of seed state is not yet implemented here

Reply:
Yes, definitely a weak spot.